### PR TITLE
Fully annotate eolearn.features

### PR DESCRIPTION
**Pull request by zigaLuksic**

### CODE REVIEW

File: features/eolearn/features/hog.py [hunk 1]
```diff
@@ -27,32 +30,26 @@ class HOGTask(EOTask):
...
-        :type feature: (FeatureType, str) or (FeatureType, str, str)
         :param orientations: Number of direction to use for the oriented gradient
-        :type orientations: int
         :param pixels_per_cell: Number of pixels in a cell
-        :type pixels_per_cell: (int, int)
         :param cells_per_block: Number of cells in a block
-        :type cells_per_block: (int, int)
         :param visualize: Produce a visualization for the HOG in an image
-        :type visualize: bool
         :param visualize_feature_name: Name of the visualization feature to be added to the eopatch (if empty and
```
Comment (jgersak) [inline features/eolearn/features/hog.py:51]: i think '(' is redundant or put ')' at the end of line 52

File: features/eolearn/features/hog.py [hunk 2]
```diff
@@ -27,32 +30,26 @@ class HOGTask(EOTask):
...
         :param feature: A feature that will be used and a new feature name where data will be saved. If new name is not
             specified it will be saved with name '<feature_name>_HOG'
 
             Example: `(FeatureType.DATA, 'bands')` or `(FeatureType.DATA, 'bands', 'hog')`
-        :type feature: (FeatureType, str) or (FeatureType, str, str)
         :param orientations: Number of direction to use for the oriented gradient
-        :type orientations: int
         :param pixels_per_cell: Number of pixels in a cell
-        :type pixels_per_cell: (int, int)
         :param cells_per_block: Number of cells in a block
```
Comment (jgersak) [inline features/eolearn/features/hog.py:49]: i think docs are ambiguous, but i do not know how to clarify them

File: features/eolearn/features/doubly_logistic_approximation.py [hunk 3]
```diff
@@ -9,53 +9,50 @@
...
-    # pylint: disable=invalid-name,locally-disabled
-    """
-    Function that is passed to `scipy.optimize`
-    """
-    return initial_value + scale * np.piecewise(
-        middle,
-        [middle < a1, middle >= a1],
-        [lambda y: np.exp(-(((a1 - y) / a4) ** a5)), lambda y: np.exp(-(((y - a1) / a2) ** a3))],
-    )
+def doubly_logistic(middle, initial_value, scale, a1, a2, a3, a4, a5) -> np.ndarray:
```
Comment (jgersak) [inline features/eolearn/features/doubly_logistic_approximation.py:21]:
```suggestion
def doubly_logistic(
    middle: float, initial_value: float, scale: float, a1: float, a2: float, a3: float, a4: float, a5: float
) -> np.ndarray:
```

Comment (zigaLuksic) [reply features/eolearn/features/doubly_logistic_approximation.py:21]: I planned to do thins in the original. But I am very sure some of these are meant to be numpy.ndarrays. It is not clear which ones or if any

File: features/eolearn/features/hog.py (hunk 1 above)
Comment (zigaLuksic) [reply features/eolearn/features/hog.py:51]: done

**Review by jgersak — APPROVED** [review]

**Pull request merged**